src/Encamina.Enmarcha.SemanticKernel/MemoryStoreExtender.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Ignore Spelling: Upsert
+
+using System.Collections.ObjectModel;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+
+using Encamina.Enmarcha.SemanticKernel.Abstractions;
+using Encamina.Enmarcha.SemanticKernel.Abstractions.Events;
+
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+
+namespace Encamina.Enmarcha.SemanticKernel;
+
+/// <summary>
+/// Extender that provides some CRUD operations over memories with multiple chunks that need to be extended by an <see cref="IMemoryStore"/>, using batch operations.
+/// </summary>
+public class MemoryStoreExtender : IMemoryStoreExtender
+{
+    private const string ChunkSize = @"chunkSize";
+
+    private readonly ILogger logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MemoryStoreExtender"/> class.
+    /// </summary>
+    /// <param name="memoryStore">A valid instance of a <see cref="IMemoryStore"/> to extend.</param>
+    /// <param name="logger">Log service.</param>
+    public MemoryStoreExtender(IMemoryStore memoryStore, ILogger<MemoryStoreExtender> logger)
+    {
+        MemoryStore = memoryStore;
+        this.logger = logger;
+    }
+
+    /// <inheritdoc cref="IMemoryStoreExtender.MemoryStoreEvent" />
```

Comment:
Why the `inheritdoc` has a `cref`?
It is not necessary 🤨

Reply:
Thank you for your contribution. You are right. I will update it.